### PR TITLE
Added dotnet isolated support. Added dotnet 5.0 as in preview.

### PR DESCRIPTION
**Pull request by gzuber**
Had to make some guesses on the best way to represent this, but wanted to get something out that we can iterate on. Biggest questions for me:

- Should this just be another `majorVersion` for dotnet? I thought it was sufficiently different that it should be separate, but I don't know if the plan is to replace the old dotnet completely.
- Since it was the only `majorVersion` of dotnet isolated, I had to make it both `isPreview` and 'isDefault`. If it moves to be in normal dotnet, it shouldn't be default.
- Guessed on the `display` and `displayVersion` fields

### CODE REVIEW

**Comment by gzuber** [comment]
@Francisco-Gamino @ammanifold for discussion

**Comment by ammanifold** [comment]
Hey @gzuber I've already merged this in on our repro, can you please check out the stacks there?

File: WindowsFunctionsStacks.json
Comment (pragnagopa) [inline WindowsFunctionsStacks.json:24]: @ankitkumarr - .Net 5 framework site config?

File: LinuxFunctionsStacks.json
Comment (pragnagopa) [inline LinuxFunctionsStacks.json:25]: @ankitkumarr - Please ACK the change.

**Comment by pragnagopa** [comment]
@Ankitkumarr - can you please link issue in AZ CLI , VS Code , Ux and Az Cmdlets here that explains differences in setting Linux Consumption vs Linux Dedicated?

**Comment by ankitkumarr** [comment]
@gzuber, as Allison mentioned, we can use this source for all the settings -- https://github.com/Azure/azure-functions-ux/blob/dev/server/src/stacks/2020-10-01/stacks/function-app-stacks/Dotnet.ts

File: WindowsFunctionsStacks.json
Comment (ankitkumarr) [reply WindowsFunctionsStacks.json:24]: Yeah Windows requires `"netFrameworkVersion": "v5.0"` site config here

**Comment by pragnagopa** [comment]
@ankitkumarr / @ammanifold  - thanks for the pointers. We do need to keep this repo in sync until we have ARM API

**Comment by ammanifold** [comment]
> @ankitkumarr / @ammanifold - thanks for the pointers. We do need to keep this repo in sync until we have ARM API

Yeah absolutely. Just wanted to give the pointer that we already have this merged in, so should be easy to get exact parity.

File: LinuxFunctionsStacks.json
Comment (ammanifold) [inline LinuxFunctionsStacks.json]: Please update these fields to match https://github.com/Azure/azure-functions-ux/blob/dev/server/src/stacks/2020-10-01/stacks/function-app-stacks/Dotnet.ts
Thanks!

**Comment by gzuber** [comment]
Thanks everyone! After speaking with @ankitkumarr we decided this should go under the "dotnet" runtime instead of making a separate "dotnet-isolated" runtime. So, with things the way they are a customer would make a dotnet 5 app like so --

```
az functionapp create --runtime dotnet --runtime-version 5.0
```

5.0 should be displayed as "preview" in the help menu. 

@ammanifold thanks for linking your stacks that helped a lot, I think I matched everything I could. Unfortunately there are some things this version of stacks doesn't have that I couldn't fit in (like display names for specific version of the runtime). But that should get resolved when the ARM API is released and we change the CLI to use it.

**Review by pragnagopa — APPROVED** [review]
LGTM! But please wait for ack from @ankitkumarr  - as the values are different for linux consumption vs dedicated

**Comment by gzuber** [comment]
> LGTM! But please wait for ack from @ankitkumarr - as the values are different for linux consumption vs dedicated

I believe I spoke to him about that and since it's temporary, I added a bit of code in the CLI to account for that difference (emptying the linuxFxVersion for consumption). But I'll wait for confirmation

**Comment by pragnagopa** [comment]
Merging this PR as the comment is resolved offline.

**Comment by ankitkumarr** [comment]
@gzuber, thank you! This all looks good. And a note on the create experience from Azure CLI. I just had a conversation with @anthonychu and others, and it sounds like we want to go with `az functionapp create --runtime dotnet-isolated --runtime-version 5.0` (opposed to `--runtime dotnet`), which you had originally proposed. This makes it a clear separation between the current in-proc `dotnet` vs `dotnet-isolated` (that's what we are calling this everywhere anyway).
Thanks for all these changes!

**Pull request merged**
